src/engine/Server.cpp

```diff
@@ -437,45 +438,48 @@ CPP_template_2(typename RequestT, typename ResponseT)(
       // sent to the client already. We can stop here.
       co_return;
     }
-    ad_utility::websocket::MessageSender messageSender = createMessageSender(
-        queryHub_, request, parsedOperation._originalString);
+    ad_utility::websocket::MessageSender messageSender =
+        createMessageSender(queryHub_, request, operationString);
 
     auto [qec, cancellationHandle, cancelTimeoutOnDestruction] =
-        prepareOperation(operationName, parsedOperation._originalString,
-                         messageSender, parameters, timeLimit.value());
-    if (!expectedOperation(parsedOperation)) {
-      throw std::runtime_error(
-          absl::StrCat(msg, ad_utility::truncateOperationString(
-                                parsedOperation._originalString)));
+        prepareOperation(operationName, operationString, messageSender,
+                         parameters, timeLimit.value());
+    if (!ql::ranges::all_of(operations, expectedOperation)) {
+      throw std::runtime_error(absl::StrCat(
+          msg, ad_utility::truncateOperationString(operationString)));
     }
-    if (parsedOperation.hasUpdateClause()) {
+    if (ql::ranges::all_of(operations,
+                           std::mem_fn(&ParsedQuery::hasUpdateClause))) {
```

Comment:
Is the `std::mem_fn` necessary here, does then plain `&hasUpdateClause` not work?